### DISCUSSION TITLE
Meaning of `signed` and `unsigned`

### DISCUSSION THREAD

**Discussion by martinberger** [rems-project/sail#349] [General]
What is the exact meaning of the `signed` and `unsigned` functions? Where are they defined or documented?

**Comment by Timmmm** [comment]
For completeness (I think you probably figure this out already!), they're defined [here](https://github.com/rems-project/sail/blob/ba9f7f1e755ea838bf5fa457937e4ed6efe8c8bb/lib/vector.sail#L356-L374) and they convert `bits(N)` to `range(0, 2^N-1)` or `range(-2^(N-1), 2^(N-1)-1)` for `unsigned` and `signed` respectively.

**Comment by martinberger** [comment]
Thanks. Yes I know this. The intention of my post was gently to suggest documenting `signed` and `unsigned` for the benefit of Sail learners.
